k6-tests/src/main.rs

```diff
@@ -0,0 +1,237 @@
+            .env("GRAPH_NODE_ADMIN_URI", test_setup.graph_node_admin_uri())
+            .env(
+                "GRAPH_NODE_HTTP_PORT",
+                test_setup.graph_node_ports.http.to_string(),
+            )
+            .env(
+                "GRAPH_NODE_INDEX_PORT",
+                test_setup.graph_node_ports.index.to_string(),
+            )
+            .env("IPFS_URI", &test_setup.ipfs_uri)
+            .current_dir("./k6-tests/subgraph")
+            .output()
+            .await
+            .context("failed to run test command")?;
+
+        println!(
+            "deploy_and_seed: stdout: {}",
+            String::from_utf8_lossy(&output.stdout)
+        );
+
+        Ok(())
+    }
+
+    async fn spawn_graph_node(&mut self) -> anyhow::Result<IntegrationTestSetup> {
+        let endpoints = self.endpoints().await?;
+        let test_setup = IntegrationTestSetup {
+            postgres_uri: endpoints.postgres,
+            ipfs_uri: endpoints.ipfs,
+            ganache_uri: endpoints.ganache.1,
+            ganache_port: endpoints.ganache.0,
+            graph_node_bin: Arc::new(self.graph_node_bin_path()),
+            graph_node_ports: GraphNodePorts::get_ports(),
+            test_directory: None,
+        };
+
+        println!("test_setup: {:?}", test_setup);
+
+        self.graph_node = Some(run_graph_node(&test_setup).await?);
+
+        Ok(test_setup)
+    }
+
+    async fn stop_graph_node(&mut self) -> anyhow::Result<()> {
+        if let Some(graph_node_child) = &mut self.graph_node {
+            let graph_node_stdio = stop_graph_node(graph_node_child).await?;
+            println!("graph_node_stdio: {:?}", graph_node_stdio);
+            self.graph_node = None;
+        }
+
+        Ok(())
+    }
+
+    async fn stop_containers(&self) -> anyhow::Result<()> {
+        self.postgres
+            .stop()
+            .await
+            .context("failed to stop container service for PostgreSQL")?;
+
+        self.ipfs
+            .stop()
+            .await
+            .context("failed to stop container service for IPFS")?;
+
+        self.ganache
+            .stop()
+            .await
+            .context("failed to stop container service for Ganache")?;
+
+        Ok(())
+    }
+}
+
+fn sleep(secs: u64) {
+    thread::sleep(time::Duration::from_secs(secs));
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    println!("Preparing to run query performance testing with K6...");
+    let mut test_env = PerformanceTestingEnvrionment::create_and_start().await?;
+
+    println!("Waiting for containers to be ready...");
+    test_env.containers_ready().await?;
+
+    println!("Creating databases...");
+    test_env.create_db().await?;
+
+    println!("Running graph-node in background...");
+    let cfg = test_env.spawn_graph_node().await?;
+    sleep(10);
+
+    println!("Seeding Smart Contract and Subgraph...");
+    test_env.deploy_and_seed(&cfg).await?;
+    sleep(10);
```

Comment:
Have you tried different values or 10 seconds is the most reliable option?

Reply:
Seems like for CI it takes ~7s and locally (m1) ~5s, I think 10s should be fine?